### PR TITLE
2 - News Page & Events Page - Shayan

### LINKED ISSUE #2: News - Page & Events - Page
**As A** volunteer/patient/stakeholder 
**I WANT TO** be a able to visit an events page and news page
**SO THAT** I can learn more about what News are scheduled by the organisation

### Acceptance Criteria

- [x] Should display a list of hardcoded News cards
- [x] Each card should contain an image and some text
- [x] Should provide a basic search/filter
- [x] ~User should be able to paginate through cards~

### PR DESCRIPTION
**Pull request by ShayanMahnam**
## Important 

Do I need to add Dates now?

## Description

- Create ContentBox component in global components
- Reuse the ContentBox component for both 'News' and 'Events' pages
- Add search functionality in the News page and show how many News found
- Add search functionality in the Events page and show how many News found
- Add select options for the Events page 

## Related to

Issue #2 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] ~I have commented my code~ (nothing complex)
- [x] ~I have updated any documentation~(no need)

### CODE REVIEW

File: client/src/components/Share/ContentBox.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		<section className="content-box">
+			<div className="image-wrapper">
+				<img className="content-img" src={image} alt="news" />
+			</div>
+			<div className="contents">
+				<h3 className="content-title">{title}</h3>
+				<p className="content-text">{truncatedContent}</p>
+				<button className="content-button">Read More</button>
+			</div>
+		</section>
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.jsx]:
```suggestion
		<section className="content-box">
			<div className="content-box-image-container">
				<img className="content-box-image" src={image} alt={altText} />
			</div>
			<div className="content-box-subcontainer">
				<h3 className="content-box-title">{title}</h3>
				<p className="content-box-text">{truncatedContent}</p>
				<button className="content-box-button">Read More</button>
			</div>
		</section>
```

File: client/src/components/Share/ContentBox.css [hunk 2]
```diff
@@ -0,0 +1,67 @@
+.content-box {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    background-color: var(--gray80);
+    width: 350px;
+    margin: 0 0 2rem 0;
+    position: relative;
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]: I am interested to learn more CSS 
And want to know what relative is doing in this case?
Could you explain it 👍

File: client/src/components/Share/ContentBox.css [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+.image-wrapper {
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]:
```suggestion
.content-box-image-container {
```

File: client/src/components/Share/ContentBox.css [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  width: 350px;
+  height: 250px;
+  overflow: hidden;
+  margin-bottom: .5rem;
+}
+
+.content-img {
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]:
```suggestion
.content-box-image {
```

File: client/src/components/Share/ContentBox.css [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+}
+
+.contents{
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]:
```suggestion
.content-box-subcontainer {
```

File: client/src/components/Share/ContentBox.css [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    width: 70%;
+    gap: .5rem;
+}
+.content-button{
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]:
```suggestion
.content-box-button {
```

File: client/src/components/Share/ContentBox.css [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  align-self: flex-end;
+  margin-bottom: .5rem;
+  padding: .3rem .5rem;
+  background-color: var(--gray60);
+  border: 1px solid var(--black);
+  color: var(--white);
+  cursor: pointer;
+}
+
+.content-button:hover{
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]:
```suggestion
.content-box-button:hover {
```

File: client/src/components/Share/ContentBox.css [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+.content-box {
+    flex-direction: row;
+    gap: 2rem;
+    align-items: start;
+    width: auto;
+    height: 250px;
+    
+}
+.contents{
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]:
```suggestion
.content-box-subcontainer {
```

File: client/src/components/Share/ContentBox.css [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    margin: 1rem 2rem 2rem 0;
+}
+.content-button{
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]:
```suggestion
.content-box-button {
```

File: client/src/components/Share/ContentBox.css [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  position: absolute;
+  bottom: 8px;
+}
+.content-title{
```
Comment (bazmurphy) [inline client/src/components/Share/ContentBox.css]:
```suggestion
.content-box-title {
```

File: client/src/pages/EventsPage/EventsPage.jsx [hunk 11]
```diff
@@ -1,11 +1,94 @@
...
+					onChange={handleSearch}
+				/>
+				<select value={selectedCategory} onChange={handleCategoryChange}>
+					<option value="All">All</option>
+					<option value="Emergency">Emergency</option>
+					<option value="Surgical">Surgical</option>
+					<option value="Diagnostic">Diagnostic</option>
+				</select>
+				<p>{filteredEvents.length} events found</p>
+			</div>
```
Comment (bazmurphy) [inline client/src/pages/EventsPage/EventsPage.jsx]: for accessibility let's make this a form

```suggestion
			<form className="search-form">
				<input
					type="text"
					placeholder="Search"
					value={searchValue}
					onChange={handleSearch}
				/>
				<select value={selectedCategory} onChange={handleCategoryChange}>
					<option value="All">All</option>
					<option value="Emergency">Emergency</option>
					<option value="Surgical">Surgical</option>
					<option value="Diagnostic">Diagnostic</option>
				</select>
				<p>{filteredEvents.length} events found</p>
			</form>
```

File: client/src/pages/NewsPage/NewsPage.jsx [hunk 12]
```diff
@@ -1,11 +1,75 @@
...
+			<Line />
+			<div className="tools-bar">
+				<input
+					type="text"
+					placeholder="Search"
+					value={searchValue}
+					onChange={handleSearch}
+				/>
+				<p>{filteredNews.length} news found</p>
+			</div>
```
Comment (bazmurphy) [inline client/src/pages/NewsPage/NewsPage.jsx]: for accessibility let's make this a form

```suggestion
			<form className="search-form">
				<input
					type="text"
					placeholder="Search"
					value={searchValue}
					onChange={handleSearch}
				/>
				<p>{filteredNews.length} news found</p>
			</form>
```

File: client/src/pages/EventsPage/EventsPage.jsx [hunk 13]
```diff
@@ -1,11 +1,94 @@
...
+			<div className="cards-list-container">
+				{filteredEvents.map((item, index) => {
+					return (
+						<ContentBox
+							key={index}
+							title={item.title}
+							image={item.image}
+							content={item.description}
+						/>
```
Comment (bazmurphy) [inline client/src/pages/EventsPage/EventsPage.jsx]: addition of altText

```suggestion
						<ContentBox
							key={index}
							title={item.title}
							image={item.image}
							altText={item.altText}
							content={item.description}
						/>
```

File: client/src/pages/NewsPage/NewsPage.jsx [hunk 14]
```diff
@@ -1,11 +1,75 @@
...
+			<div className="cards-list-container">
+				{filteredNews.map((item, index) => {
+					return (
+						<ContentBox
+							key={index}
+							title={item.title}
+							image={item.image}
+							content={item.description}
+						/>
```
Comment (bazmurphy) [inline client/src/pages/NewsPage/NewsPage.jsx]: addition of altText

```suggestion
						<ContentBox
							key={index}
							title={item.title}
							image={item.image}
							altText={item.altText}
							content={item.description}
						/>
```

File: client/src/pages/EventsPage/EventsPage.jsx [hunk 15]
```diff
@@ -1,11 +1,94 @@
...
+			.toLowerCase()
+			.includes(searchValue.toLowerCase());
+		const contentMatches = item.description
+			.toLowerCase()
+			.includes(searchValue.toLowerCase());
+		const categoryMatches =
+			selectedCategory === "All" || item.category === selectedCategory;
+
+		return (titleMatches || contentMatches) && categoryMatches;
+	});
```
Comment (bazmurphy) [inline client/src/pages/EventsPage/EventsPage.jsx]:
```suggestion
	const filteredEvents = eventsData.filter((item) => {
		const titleMatches = item.title
			.toLowerCase()
			.includes(searchValue.toLowerCase());
		const descriptionMatches = item.description
			.toLowerCase()
			.includes(searchValue.toLowerCase());
		const categoryMatches =
			selectedCategory === "All" || item.category === selectedCategory;

		return (titleMatches || descriptionMatches ) && categoryMatches;
	});
```

File: client/src/pages/NewsPage/NewsPage.jsx [hunk 16]
```diff
@@ -1,11 +1,75 @@
...
+
+	const filteredNews = newsData.filter((item) => {
+		const titleMatches = item.title
+			.toLowerCase()
+			.includes(searchValue.toLowerCase());
+		const contentMatches = item.description
+			.toLowerCase()
+			.includes(searchValue.toLowerCase());
+		return titleMatches || contentMatches;
+	});
```
Comment (bazmurphy) [inline client/src/pages/NewsPage/NewsPage.jsx]:
```suggestion
	const filteredNews = newsData.filter((item) => {
		const titleMatches = item.title
			.toLowerCase()
			.includes(searchValue.toLowerCase());
		const descriptionMatches = item.description
			.toLowerCase()
			.includes(searchValue.toLowerCase());
		return titleMatches || descriptionMatches;
	});
```

File: client/src/pages/EventsPage/EventsPage.jsx [hunk 17]
```diff
@@ -1,11 +1,94 @@
...
+import ContentBox from "../../components/Share/ContentBox";
+import Line from "../../components/Line/Line";
+import { useState } from "react";
+
+const eventsData = [
+	{
+		title: "Emergency Services",
+		image:
+			"https://clevelandcliniclondon.uk/-/scassets/images/org/locations/london/hospital-services/hospital-services-feature.jpg",
+		description:
```
Comment (bazmurphy) [inline client/src/pages/EventsPage/EventsPage.jsx:12]: as an example:
```suggestion
    altText: "some alt text for the image"
		description:
```

File: client/src/pages/EventsPage/EventsPage.css [hunk 18]
```diff
@@ -0,0 +1,13 @@
+.tools-bar{
```
Comment (bazmurphy) [inline client/src/pages/EventsPage/EventsPage.css]:
```suggestion
.search-form {
```

File: client/src/pages/NewsPage/NewsPage.css [hunk 19]
```diff
@@ -0,0 +1,14 @@
+.tools-bar{
```
Comment (bazmurphy) [inline client/src/pages/NewsPage/NewsPage.css]:
```suggestion
.search-form {
```

**Review by bazmurphy — COMMENTED** [review]
Good job 😄 

Here is my Review 😎

- div to form element change for accessibility
- sorting function variable renames
- class renames to be coherent
- addition of altText for dynamic Content Box for accessibility

(It will be neccessary to update the mock data with altText key/values to reflect the altText for the image - i gave one example)

**About folder structure:**
You can create a `ContentBox` folder and put it directly into /components/
And have the .jsx and .css in there

So /client/src/components/ContentBox/

I dont think we will need a /Share/ since `Line` is another example of a shared global component and sits directly in /components/

**About formatting:**
Whenever I look at the PR in VSCode and then save (with Prettier) it makes a lot of formatting adjustments. Are your Prettier settings working and default? Because something is not working to format the code correctly (as an example the media queries in the css file).
(otherwise we will get a lot of git diffs if everyones formatting is not working the same)

File: client/src/components/Share/ContentBox.css (hunk 2 above)
Comment (ShayanMahnam) [reply client/src/components/Share/ContentBox.css]: So here we have a text which we need to be responsive 
and when we change the size of the screen the hole content shape of the box going to change so for solve the problem I made the box relative and the readme btn absolute so the button place never going to change

**Comment by bazmurphy** [comment]
The formatter in your local environment still seems to not be working, see the two screenshots of the CSS file, directly pulled from this branch, with before and after I click Save. (Everything is indented correctly and there is an extra line at the end of the code, this is how you know Prettier is working)

Before:
image

After:
image

File: client/src/components/ContentBox/ContentBox.css [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+    margin: 1rem 2rem 2rem 0;
+}
+.content-box-button{
+  position: absolute;
+  bottom: 8px;
+}
+.content-box-title{
+  align-self: flex-start;
+}
+}
```
Comment (bazmurphy) [inline client/src/components/ContentBox/ContentBox.css:67]:
```suggestion
This is just AN EXAMPLE to highlight the formatting issue, please do not accept suggestion in this case.

@media screen and (min-width: 768px) {
	.content-box {
		flex-direction: row;
		gap: 2rem;
		align-items: start;
		width: auto;
		height: 250px;
	}
	.content-box-subcontainer {
		margin: 1rem 2rem 2rem 0;
	}
	.content-box-button {
		position: absolute;
		bottom: 8px;
	}
	.content-box-title {
		align-self: flex-start;
	}
}

```

**Review by bazmurphy — COMMENTED** [review]
The formatter in your local environment still seems to not be working
As an example, see the attached screenshots of the CSS file, directly pulled from this branch
With before and after I click Save. 
Notice: Everything gets indented correctly and there is an extra line at the end of the code.. so we see Prettier is working

**BEFORE:**
image

**AFTER:**
image

File: client/src/components/ContentBox/ContentBox.css [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+
+.content-box-subcontainer {
+	display: flex;
+	flex-direction: column;
+	justify-content: center;
+	align-items: center;
+	width: 70%;
+	gap: 0.5rem;
+}
+.content-box-button {
```
Comment (bazmurphy) [inline client/src/components/ContentBox/ContentBox.css]:
```suggestion
	gap: 0.5rem;
}

.content-box-button {
```

File: client/src/components/ContentBox/ContentBox.css [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+
+@media screen and (min-width: 768px) {
+	.content-box {
+		flex-direction: row;
+		gap: 2rem;
+		align-items: start;
+		width: auto;
+		height: 250px;
+	}
+	.content-box-subcontainer {
```
Comment (bazmurphy) [inline client/src/components/ContentBox/ContentBox.css]:
```suggestion
	}
	
	.content-box-subcontainer {
```

File: client/src/components/ContentBox/ContentBox.css [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+		margin: 1rem 2rem 2rem 0;
+	}
+	.content-box-button {
```
Comment (bazmurphy) [inline client/src/components/ContentBox/ContentBox.css]:
```suggestion
	}
	
	.content-box-button {
```

File: client/src/components/ContentBox/ContentBox.css [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+		position: absolute;
+		bottom: 8px;
+	}
+	.content-box-title {
```
Comment (bazmurphy) [inline client/src/components/ContentBox/ContentBox.css]:
```suggestion
	}
	
	.content-box-title {
```

File: client/src/pages/NewsPage/NewsPage.css [hunk 25]
```diff
@@ -0,0 +1,17 @@
+.news-page {
+	margin: 0 20px;
+}
+.search-form {
```
Comment (bazmurphy) [inline client/src/pages/NewsPage/NewsPage.css]:
```suggestion
}

.search-form {
```

**Review by bazmurphy — APPROVED** [review]
Final thing: could we all do spaces between each css rule, to keep it consistent for everyone, and so that we don't get unnecessary git diffs.

I approved it now, so you can merge it after those small adjustments.

😎 Good Job

**Pull request merged**